src/lib/Libtpp/tpp_util.c

```diff
@@ -2332,6 +2332,89 @@ tpp_set_logmask(long logmask)
 	tpp_log_event_mask = logmask;
 }
 
+/**
+ * @brief
+ *	Lock the mutex associated with tpp nslookup
+ *
+ * @return Error code
+ * @retval -1 - failure
+ * @retval  0 - success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ * @par MT-safe: Yes
+ *
+ */
+int
+tpp_nslookup_mutex_lock()
+{
+
+	if (pthread_mutex_lock(&tpp_nslookup_mutex) != 0)
+		return -1;
+
+	return 0;
+}
+
+/**
+ * @brief
+ *	Unlock the mutex associated with tpp nslookup
+ *
+ * @return Error code
+ * @retval -1 - failure
+ * @retval  0 - success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ * @par MT-safe: Yes
+ *
+ */
+int
+tpp_nslookup_mutex_unlock()
```

Comment:
you dont need these new routines, you can simply use tpp_lock() and pass the tpp_nslookup_mutex address to it.